File: Country124.java

public class Country124 {

	public static void main(String[] args) {
		System.out.println(new Country124().new Solution().solution(4));

	}
	class Solution {
		  public String solution(int n) {
			  StringBuffer answer = new StringBuffer();
			  int sel = n;
		      while(true ) {
		    	  int val = sel%3;
		    	  sel = sel/3;
		    	  if(val == 0)
		    	  {
		    		  answer.append("4");
		    		  if(sel>=1)
		    		  sel-=1;
		    	  }
		    	  else if(val == 2)
		    		  answer.append("2");
		    	  else if(val == 1)
		    		  answer.append("1");
		    	  if(sel/3 == 0 && sel%3 == 0)
		    		  break;
		      }
		      answer.reverse();
		      return answer.toString();
		  }
	}
}

File: Country124.cpp
#include<iostream>
#include<algorithm>
#include <string>
#include <vector>

using namespace std;

string solution(int n) {
	string result="";
	while (n != 0) {
		int q = n % 3;
		n /= 3;
		if (q == 0) n -= 1;

		switch (q)
		{
		case 0: result += "4";
			break;
		case 1: result += "1";
			break;
		case 2: result += "2";
			break;
		}
	}
	reverse(begin(result), end(result));
	return result;
}

int main() {
	auto c = solution(4);

}